backend/LexCore/Exceptions/NotFoundException.cs

```diff
@@ -1,8 +1,16 @@
+using System.Diagnostics.CodeAnalysis;
+
 namespace LexCore.Exceptions;
 
-public class NotFoundException : Exception
+public class NotFoundException(string message, string type) : Exception(message)
 {
-    public NotFoundException(string message) : base(message)
+    public static void ThrowIfNull<T>([AllowNull, NotNull] T arg)
     {
+        if (arg is null)
+            throw ForType<T>();
     }
+
+    public static NotFoundException ForType<T>() => new($"{typeof(T).Name} not found", typeof(T).Name);
```

Comment:
I really like this. NotFound exceptions by definition should pretty much always have a type. It looks like there's still quite a bit of cleanup you could do by adopting this.